nmtwizard/preprocess/prepoperator.py

```diff
@@ -194,9 +198,21 @@ def __call__(self, tu_batch):
 class Operator(object):
     """Base class for preprocessing operators."""
 
-    def __call__(self, tu_batch, process_type):
-        self._process_type = process_type
-        if process_type == ProcessType.POSTPROCESS:
+    def __init__(self, params, process_type, build_state):
+        pass
+
+
+    @property
```

Comment:
Do we need a property here ? Can't we simply make those public attributes ?

Reply:
The property makes the attribute read-only (build_operator is a special case). It's a bit more code but an additional "safety".

Reply:
Ok, thanks for explanation.